xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiServletResponse.java

```diff
@@ -31,6 +32,7 @@
 import org.apache.commons.lang3.StringUtils;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.xwiki.url.URLSecurityManager;
 
 public class XWikiServletResponse implements XWikiResponse
```

Comment:
Doing it at this level is quite a breaking change since XWikiServletResponse is not only used by xredirect but also by perfectly valid cases: for example this is going to break the OpenID Connect Authenticator.

Reply:
That's interesting, so you suggest it should be moved to the handling of xredirect parameters only? AFAIR there was no single point for that, I guess it would need a proper refactoring to be handled at a single place.

Reply:
Yes supporting this only for xredirect is not an easy fix and won't work for any extension which uses xredirect. A possibility could be to compare the `redirect` value with the `xredirect` value from the context request, but that's hackish...